scripts/irods/test/test_auth.py

```diff
@@ -252,3 +252,499 @@ def test_allowlist(self):
+        # Make sure that the authentication succeeded.
+        assert_command('ils', 'STDOUT', f'/{self.zone}/home/{self.user}')
+
+
+    @unittest.skipIf(test.settings.USE_SSL, "Basic iinit usage does not configure SSL.")
+    def test_iinit_basic_authentication_failure(self):
+        # Execute iinit with the basic prompts only, with an incorrect password.
+        cmd = 'iinit'
+        user_input = os.linesep.join(
+            [self.host, self.port, self.user, self.zone, 'nopes']) + os.linesep
+        stdout, stderr, rc = lib.execute_command_permissive(cmd, input=user_input, env=os.environ)
+        lib.log_command_result(cmd, stdout, stderr, rc)
+
+        # Ensure program did not terminate gracefully.
+        self.assertNotEqual(rc, 0)
+
+        # Check for expected error string.
+        error_string = 'CAT_INVALID_AUTHENTICATION'
+        self.assertIn(error_string, stderr)
+
+        self.assert_basic_iinit_prompts_are_in_stdout(stdout)
+        self.assertIn('Enter your current iRODS password', stdout)
+
+        # Make sure that the environment file saved despite authentication failure.
+        self.assert_environment_file_contents()
+
+        # Make sure that the authentication failed.
+        assert_command('ils', 'STDERR', error_string)
+
+
+    @unittest.skipIf(test.settings.USE_SSL, "This test does not configure SSL for the authenticated user.")
+    def test_iinit_authscheme_success_with_default_answers(self):
+        # Execute iinit with the basic prompts only, with an incorrect password.
+        cmd = 'iinit --prompt-auth-scheme'
+        user_input = os.linesep.join([
+            self.host,
+            '', # port
+            self.user,
+            self.zone,
+            '', # authentication scheme
+            self.password]) + os.linesep
+        stdout, stderr, rc = lib.execute_command_permissive(cmd, input=user_input, env=os.environ)
+        lib.log_command_result(cmd, stdout, stderr, rc)
+
+        # Ensure program terminated gracefully.
+        self.assertEqual(rc, 0)
+
+        self.assert_basic_iinit_prompts_are_in_stdout(stdout)
+        self.assertIn('Enter your current iRODS password', stdout)
+        self.assert_auth_scheme_iinit_prompts_are_in_stdout(stdout)
+
+        # Make sure that the environment file saved with the correct contents.
+        self.assert_environment_file_contents(
+            expected_json_contents = self.merge_dicts(
+                self.base_client_environment, {'irods_authentication_scheme': 'native'}))
+
+        # Make sure that the authentication succeeded.
+        assert_command('ils', 'STDOUT', f'/{self.zone}/home/{self.user}')
+
+
+    @unittest.skipIf(test.settings.USE_SSL, "This test does not configure SSL for the authenticated user.")
+    def test_iinit_authscheme_failure_with_invalid_scheme(self):
+        # Execute iinit with the auth scheme prompt and provide a valid scheme that will fail with default config.
+        auth_scheme = 'scheme_what_dont_exist'
```

Comment:
Is this supposed to say "scheme_**what**_dont_exist"?

Reply:
Done

Reply:
Oh, I see now lol. I'll keep that in mind next time.